src/overlays/actors/ovl_En_Kaizoku/z_en_kaizoku.c

```diff
@@ -65,118 +136,1921 @@ const ActorInit En_Kaizoku_InitVars = {
+        if (player->meleeWeaponAnimation == 0x11) {
+            if (this->unk_2B0 != 2) {
+                func_80B874D8(this, play);
+            }
+            return 1;
+        } else {
+            func_80B87900(this);
+            return 1;
+        }
+    } else if (func_800BE184(play, &this->actor, 100.0f, 0x5DC0, 0x2AA8, this->actor.shape.rot.y)) {
+        this->actor.shape.rot.y = this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+        if ((this->actor.bgCheckFlags & 8) && ABS_ALT(sp32) < 12000 && this->actor.xzDistToPlayer < 90.0f) {
+            if (this->unk_2B0 != 6) {
+                func_80B87C7C(this);
+            }
+            return 1;
+        } else if (player->meleeWeaponAnimation == 17) {
+            if (this->unk_2B0 != 2) {
+                func_80B874D8(this, play);
+            }
+        } else if (this->actor.xzDistToPlayer < gGameInfo->data[0x96B] + 180.0f) {
+            func_80B87900(this);
+        } else if (this->unk_2B0 != 7) {
+            func_80B87E28(this);
+        }
+        return 1;
+    }
+
+    sp2C = Actor_FindNearby(play, &this->actor, -1, ACTORCAT_EXPLOSIVES, 80.0f);
+    if (sp2C != NULL) {
+        this->actor.shape.rot.y = this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+        if (((this->actor.bgCheckFlags & 8) && sp32 < 12000) || sp2C->id == ACTOR_EN_BOM_CHU) {
+            if (sp2C->id == ACTOR_EN_BOM_CHU && Actor_DistanceBetweenActors(&this->actor, sp2C) < 80.0f &&
+                (s16)(this->actor.shape.rot.y - sp2C->world.rot.y + 0x8000) < 0x4000) {
+                if (this->unk_2B0 != 6) {
+                    func_80B87C7C(this);
+                }
+            } else {
+                func_80B87900(this);
+            }
+            return 1;
+        }
+        if (this->unk_2B0 != 7) {
+            func_80B87E28(this);
+        }
+        return 1;
+    }
+
+    if (arg2) {
+        s16 sp2A;
+    label:;
+        if (sp30 >= 10000) {
+            func_80B87900(this);
+        } else {
+            sp2A = player->actor.shape.rot.y - this->actor.shape.rot.y;
+            if (this->actor.xzDistToPlayer <= 65.0f && !Actor_OtherIsTargeted(play, &this->actor) &&
+                ABS_ALT(sp2A) < 0x5000) {
+                if (this->unk_2B0 != 9) {
+                    func_80B87F70(this);
+                    return 1;
+                }
+            } else if (this->unk_2B0 != 3) {
+                func_80B88CD8(this);
+            }
+        }
+        return 1;
+    }
+    return 0;
+}
+
+extern u32 D_80B8ABF8[];
+extern u8 D_80B8AC44[];
+
+void func_80B85E18(EnKaizoku* this, s32 arg1) {
+    this->unk_2E4 = arg1;
+    this->unk_2DC = Animation_GetLastFrame(D_80B8ABF8[this->unk_2E4]);
+    Animation_Change(&this->skelAnime, D_80B8ABF8[this->unk_2E4], 1.0f, 0.0f, this->unk_2DC, D_80B8AC44[this->unk_2E4],
+                     0.0f);
+}
+
+s32 func_80B85EA0(PlayState* play, Actor* actor) {
```

Comment:
Just going to note for myself for later that this is a `PictoActor` and this is a verification function.

Reply:
Followup: now that #952 is in, can you switch this over to a PictoActor and follow the function naming scheme I established in EnGe1?